torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -976,6 +976,7 @@ class AsyncSparseAllreduceWork : public ProcessGroupGloo::AsyncWork {
     // Copy back to input tensors.
     outputs.reserve(inputs.size());
     for (size_t i = 0; i < inputs.size(); i++) {
+      inputs[i].copy_(output);
```

Comment:
I am not removing outputs right now, because it is used in result() API in line 988. 

@pietern I'm wondering why we want to differentiate output.is_sparse() or not in line 980 while copying back output to outputs

Reply:
@ifedan and @VitalyFedyunin added this. I suspect it is because sparse tensors are actually two tensors, and therefore it does not support contiguous memory format?

Reply:
It's because all other primitives supply the output tensors as argument and treat the input as const. You can't know the shape of a sparse output tensor in advance, so we chose to return it out of band through the work object.